ecosystem/sep-0039.md

```diff
@@ -0,0 +1,193 @@
+
+### NFT Issuance
+Given an arbitrary digital asset, here's how issuance typically works:
+
+  1. An account is created to represent the digital asset on the ledger
+  2. The account stores the entire digital asset within its managed data entries.
+  3. The account issues a Stellar asset representing ownership over the digital asset we just stored.
+  4. Any account holding that Stellar asset lays claim to the digital asset.
+
+Notice that ownership of the token directly conveys ownership of a unit of the asset, entirely within the Stellar ledger. Other scenarios such as locking down the issuing account to avoid supply increases, ownership transfers via payments, etc. are built on top of this basic model and are thus extraneous to this SEP.
+
+Our focus is Step 2: storing the asset within the account.
+
+### Encoding Binary Data
+We begin by describing how binary data should be represented in a `ManageData` entry.
+
+Since the _key_ is restricted specifically to printable characters (ref: Stellar Core's [`isString32Valid()`](https://github.com/stellar/stellar-core/blob/b031954458df3bb31d8d98f136c6fee40523a10d/src/util/types.cpp#L89-L101) and C's [`isprint()`](https://en.cppreference.com/w/cpp/string/byte/isprint)), this gives us an encoding space of 94 characters or ~6.5 bits per character (a ~25% reduction in the space). From some cursory research, the [BasE91] encoding scheme closest to this theoretical limit and outperforms base64 encoding.
+
+The _value_ can be any binary data; thus, no encoding is necessary.
+
+### Representing the Data
+With the encoding described [above](#encoding-binary-data), we can now discuss the actual storage. Again, we'll use the `ManageData` operation to store the binary data. 
+
+The first entry _may_ be a [MIME type], followed by the first chunk of data, separated by a comma character (`,`). The MIME type *must* fit in a single key, and so must be <= 63 characters long.
+
+If the MIME type is excluded, the comma _must_ still be included so that readers can easily distinguish where the type starts and the data begins.
+
+Every subsequent entry is simply the next BasE91-encoded chunk, sized to fit into the 64-length key string, and the next 64-byte binary chunk.
+
+If the final entry fits into the key (that is, the value is not necessary), the encoder _must_ split it into a key and a value. Specifically, since the number of bytes remaining is definitely <= 64, the key _must_ be composed of dash (`-`) characters and the value _must_ be the final chunk.
+
+For example,
+
+| Key | Value |
+|:----|:-----:|
+| text/plain,'ZP^gp@..<etc>.. | .. 64 bytes .. |
+| Io1Tc%ZEYkE#^IR`0e..<etc>.. | .. 64 bytes .. |
+| %yO){B&mA#_1:W8{lT..<etc>.. | .. 64 bytes .. |
+| Cvnu8jZBP>1]TX2$g%..<etc>.. | .. 64 bytes .. |
+| `-`                         | B' |
```

Comment:
As mentioned in my comment earlier you can have empty values. They just need to be empty Buffers. e.g. `Buffer.alloc(0)` for example.